reconcile/utils/acs/policies.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from reconcile.utils.acs.base import AcsBaseApi
+
+
+class Scope(BaseModel):
+    cluster: str
+    namespace: Optional[str]
+
+
+class PolicyCondition(BaseModel):
+    field_name: str
+    negate: bool
+    values: list[str]
+
+
+class Policy(BaseModel):
+    name: str
+    description: str
+    notifiers: list[str]
+    categories: list[str]
+    severity: str
+    scope: list[Scope]
+    conditions: list[PolicyCondition]
+
+
+class AcsPolicyApi(AcsBaseApi):
+    def list_custom_policies(self) -> list[Any]:
+        # retrieve summary data for each custom policy
+        return [
+            p
+            for p in self.generic_request("/v1/policies", "GET").json()["policies"]
+            if not p["isDefault"]
+        ]
+
+    def get_custom_policies(self) -> list[Policy]:
+        custom_policy_ids = [p["id"] for p in self.list_custom_policies()]
+        # make individual policy requests to obtain further details
+        custom_policies_api_result = [
+            self.generic_request(f"/v1/policies/{pid}", "GET").json()
+            for pid in custom_policy_ids
+        ]
```

Comment:
This will send multiple requests, is there an API to batch return policy details?

Reply:
unfortunately not. And there are several attributes not returned by the bulk `/v1/policies` request that necessitates making the individual `/v1/policies/{id}` requests.

Reply:
It's not detailed well in the docs but `/v1/policies` returns
```json
    {
      "id": "string",
      "name": "string",
      "description": "string",
      "severity": "string",
      "disabled": false,
      "lifecycleStages": [
        "string"
      ],
      "notifiers": [],
      "lastUpdated": "string",
      "eventSource": "string",
      "isDefault": false
    }
```

whereas `/v1/policies/{some_policy_id}` yields additional attributes for the policy. The noteworthy ones for this integration being `scope` and `policySections`.